nav2_costmap_2d/plugins/costmap_filters/costmap_filter.cpp

```diff
@@ -0,0 +1,153 @@
+/*********************************************************************
+ *
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2020 Samsung Research Russia
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the <ORGANIZATION> nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+#include "nav2_costmap_2d/costmap_filters/costmap_filter.hpp"
+
+#include <exception>
+
+namespace nav2_costmap_2d
+{
+
+CostmapFilter::CostmapFilter()
+: semantic_info_topic_("/semantic_info")
+{
+}
+
+CostmapFilter::~CostmapFilter()
+{
+}
+
+void CostmapFilter::onInitialize()
+{
+  costmap_ = nullptr;
+  costmap_size_x = 0;
+  costmap_size_y = 0;
+
+  // Get parameters
+  declareParameter("enabled", rclcpp::ParameterValue(true));
+  declareParameter("semantic_info_topic", rclcpp::ParameterValue("/semantic_info"));
+
+  node_->get_parameter(name_ + "." + "enabled", enabled_);
+  node_->get_parameter(name_ + "." + "semantic_info_topic", semantic_info_topic_);
+}
+
+void CostmapFilter::activate()
+{
+  loadFilter(semantic_info_topic_);
+}
+
+void CostmapFilter::deactivate()
+{
+  unloadFilter();
+}
+
+void CostmapFilter::reset()
+{
+  if (costmap_) {
+    delete[] costmap_;
+  }
+  costmap_ = nullptr;
+  costmap_size_x = 0;
+  costmap_size_y = 0;
+
+  unloadFilter();
+  loadFilter(semantic_info_topic_);
+}
+
+void CostmapFilter::updateBounds(
```

Comment:
I don't see any support in here for rolling vs non-rolling costmaps

Reply:
Added in https://github.com/ros-planning/navigation2/commit/fcf337566456677d0a6892b8539e902dfd086994#diff-9abde77dd5850f2da94347098f064bf5R96-R100